tests/unit/experiments/providers/simple_experiment_tests.py

```diff
@@ -0,0 +1,331 @@
+        for i in range(num_users):
+            fullnames.append("t2_%s" % str(i))
+
+        counter = collections.Counter()
+        bucketing_changed = False
+        for fullname in fullnames:
+            self.assertEqual(experiment.seed, "some_new_seed")
+            bucket1 = experiment._calculate_bucket(fullname)
+            counter[bucket1] += 1
+            # Ensure bucketing is deterministic.
+            self.assertEqual(bucket1, experiment._calculate_bucket(fullname))
+
+            current_seed = experiment.seed
+            experiment._seed = "newstring"
+            bucket2 = experiment._calculate_bucket(fullname)
+            experiment._seed = current_seed
+            # check that the bucketing changed at some point. Can't compare
+            # bucket1 to bucket2 inline because sometimes the user will fall
+            # into both buckets, and test will fail. 
+            if bucket1 != bucket2:
+                bucketing_changed = True
+
+        self.assertTrue(bucketing_changed)
+
+        for bucket in range(experiment.num_buckets):
+            # We want an even distribution across buckets.
+            expected = num_users / experiment.num_buckets
+            actual = counter[bucket]
+            # Calculating the percentage difference instead of looking at the
+            # raw difference scales better as we change NUM_USERS.
+            percent_equal = float(actual) / expected
+            self.assertAlmostEqual(percent_equal, 1.0, delta=.10,
+                                   msg='bucket: %s' % bucket)
+
+    def test_variant_returns_none_if_out_of_time_window(self):
+        valid_cfg = get_simple_config()
+        experiment_valid = parse_experiment(valid_cfg)
+        experiment_valid._choose_variant = choose_variants_override
+
+        expired_cfg = get_simple_config()
+        expired_cfg['stop_ts'] = time.time() - FIVE_DAYS
+        experiment_expired = parse_experiment(expired_cfg)
+
+        experiment_not_started_cfg = get_simple_config()
+        experiment_not_started_cfg['start_ts'] = time.time() + FIVE_DAYS
+        experiment_not_started = parse_experiment(experiment_not_started_cfg)
+
+        variant_valid = experiment_valid.variant(user_id="t2_1")
+        variant_expired = experiment_expired.variant(user_id="t2_1")
+        variant_not_started = experiment_not_started.variant(user_id="t2_1")
+
+        self.assertIsNot(variant_valid, None)
+        self.assertIs(variant_expired, None)
+        self.assertIs(variant_not_started, None)
+
+    def test_no_bucket_val(self):
+        experiment = create_simple_experiment()
+        no_user_id_provided_variant = experiment.variant(not_user_id="t2_1")
+        none_user_id_provided_variant = experiment.variant(not_user_id=None)
+
+        self.assertIs(no_user_id_provided_variant, None)
+        self.assertIs(none_user_id_provided_variant, None)
+
+    def test_experiment_disabled(self):
+        experiments_cfg = get_simple_config()
```

Comment:
For this test, you might also want to change the variant sizes so the "enabled" one is at 100% to highlight that it will always be `None`.

Reply:
changed